@here/harp-mapview/lib/MapView.ts

```diff
@@ -1791,6 +1839,27 @@ export class MapView extends THREE.EventDispatcher {
         this.m_zoomLevel = MapViewUtils.calculateZoomLevelFromDistance(zoomLevelDistance, this);
     }
 
+    private detectCurrentFps(now: number) {
+        if (this.m_previousRequestAnimationTime === undefined) {
+            this.m_previousRequestAnimationTime = now;
+            return;
+        }
+
+        // Skip the first frames, they are from not originated from requestAnimationFrame()
+        if (this.m_frameNumber > 5) {
+            const currentFps = 1000 / (now - this.m_previousRequestAnimationTime);
+            this.m_frameTimeRing[this.m_frameTimeIndex] = currentFps;
+            this.m_frameTimeIndex = (this.m_frameTimeIndex + 1) % FRAME_RATE_RING_SIZE;
+
+            let sum: number = 0;
+            for (let i = 0; i < FRAME_RATE_RING_SIZE; i++) {
+                sum += this.m_frameTimeRing[i];
+            }
+            this.m_detectedFps = Math.floor(sum / FRAME_RATE_RING_SIZE);
```

Comment:
this is slightly wrong, if the ring is only partially filled, then it should divide by the number of valid entries (?)